runtime/bindings/python/hal.cc

```diff
@@ -286,6 +286,9 @@ py::object MapElementTypeToDType(iree_hal_element_type_t element_type) {
     case IREE_HAL_ELEMENT_TYPE_UINT_64:
       dtype_code = "L";
       break;
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_16:
+      dtype_code = "e";
```

Comment:
OOC reason for e? (E.g., h for half, f float, d double, could make sense too)

Reply:
We are using the numpy type codes, so sadly we don't get to define it.
https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.half